### PR TITLE
Case System

### PR DESCRIPTION
**Pull request by Maverick-Wolf**
Joe this a case system I made If you have any suggestions please do tell
And this is my first PR so no bully :(

### CODE REVIEW

File: case system/cases.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{{end}}{{end}}
+{{$skip := mult (sub $page 1) 10}}
+{{$cases := dbTopEntries $id 10 $skip}}
+{{if not $cases}}
+{{$display = "`No Cases exists on this page`"}}
+{{else}}
+{{- range $cases -}}
+		{{ $value := str .Value }} 
+		{{ $display = joinStr "" $display "\n" $value}} 
+{{- end -}}{{- end -}}
```
Comment (jo3-l2) [inline case system/cases.go.tmpl:28]:
```suggestion
{{ range $cases }}
		{{- $value := str .Value }} 
		{{- $display = joinStr "" $display "\n" $value -}} 
{{ else }}
{{$display = "`No cases exist on this page`"}}
{{ end }}
```

File: case system/viewcase.go.tmpl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{{with (dbGet $b "viewcase")}}	
+{{$a = sdict .Value}}
+	{{$name = $a.Get "name"}}
+	{{$iconurl = $a.Get "avatar"}}
+	{{$warnname = $a.Get "warnname"}}
+	{{$reason = $a.Get "reason"}}
+	{{$action = $a.Get "action"}}
+    	{{$userid = $a.Get "userid"}}
+	{{$msgid = $a.Get "msgid"}}
+	{{$channel = $a.Get "channel"}}
```
Comment (jo3-l2) [inline case system/viewcase.go.tmpl:22]: can all be transformed to the form `{{$a.Key}}` e.g. `$a.name`

File: case system/viewcase.go.tmpl [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    
+    -viewcase <case_number>
+    
+    Recommended trigger: Regex trigger with trigger `\A(-|<@!?204255221017214977>\s*)(viewcase|seecase|vc|sc|case)(\s+|\z)`
+*/}}
+{{$args := parseArgs 1 "`-viewcase case_number`" (carg "int" "case number")}}
+{{$a := ""}}{{$name := ""}}{{$iconurl := "" }}{{$warnname := "" }}{{$reason := "" }}{{$userid := ""}}{{$action := ""}}{{$link := ""}}{{$msgid := ""}}{{$channel := ""}}{{$server := .Guild.ID}}
+{{$b := ($args.Get 0)}}
+{{if (dbGet $b "viewcase")}}
+{{with (dbGet $b "viewcase")}}
```
Comment (jo3-l2) [inline case system/viewcase.go.tmpl:13]: redundant, refactor please.

File: case system/viewcase.go.tmpl [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	{{$reason = $a.Get "reason"}}
+	{{$action = $a.Get "action"}}
+    	{{$userid = $a.Get "userid"}}
+	{{$msgid = $a.Get "msgid"}}
+	{{$channel = $a.Get "channel"}}
+{{end}}
+{{$embed := cembed "author" (sdict "name" $name "url" "" "icon_url" $iconurl) "description" (print "[**Jump to Warn**](https://discordapp.com/channels/" $server "/" $channel "/" $msgid "/)\n**User** : `" $warnname "`\t`(" $userid ")`\n **Action** : `" $action "`\n **Reason** : `" $reason "`") "color" 0x00ff00 "footer" (sdict "text" (print "Case#" $b))}}
+{{sendMessage nil $embed}} 
+{{else}}
+{{sendMessage nil "Could not find the case specified,Please make sure the case number is correct or the case has not been deleted"}}
```
Comment (jo3-l2) [inline case system/viewcase.go.tmpl:27]:
```suggestion
Could not find the case specified,Please make sure the case number is correct or the case has not been deleted
```

**Review by jo3-l2 — CHANGES_REQUESTED** [review]
This currently seems to use 3 DB entries > case, can we change that to using 1 (sdict/dict entry) or even fewer?

**Comment by Maverick-Wolf** [comment]
have resolved all the other issues, about the db thingy though of doing that, but the the current use of 3 db are a little easier to edit and its just 3 db entries so its okay  :) 
c'mon 3 is small

**Comment by MatiasMFM2001** [comment]
> c'mon 3 is small

Yup, but that needs more DB operations, which are costly (10 max). No idea if that also makes CC slower, due to LAGPDB

Using a single entry, you can fetch all data at once. Even with `dbTopEntries`/etc if you want to get a lot of entries to iterate them

**Comment by EsenDR** [comment]
@Maverick-Wolf are you still working on this or do you plan to work on this? If not, maybe close this PR?

**Comment by Maverick-Wolf** [comment]
yeah will close this for now

**Pull request closed without merging**